### ISSUE TITLE
Projects can declare dependencies on plugin artifacts using plugin coordinates instead of artifact coordinates

### ISSUE THREAD

**Issue by bamboo** [gradle/kotlin-dsl#430]
```kotlin
   // buildSrc/build.gradle.kts
   plugins {
      `kotlin-dsl`
   }
   
   dependencies {
      compile(plugin("org.foo") version "1.0")
      compile(plugin("org.bar") version "2.3")
   }
```

**Comment by bamboo** [comment]
Blocked by #389.

**Comment by gildor** [comment]
I'm just curious what is rationale of this task? To use plugin without apply plugin itself?

**Comment by JLLeitschuh** [comment]
Shouldn't groovy have this too?? Or does it already?

**Comment by mkobit** [comment]
I would think this would enable easier usage of composing multiple plugins together.

**Comment by eskatos** [comment]
@mkobit, yes

Rationale is to ease declaring dependencies on Gradle plugins in binary plugin builds or `buildSrc` by making the declaration close to consuming a plugin. Allowing mechanical/automated refactoring.

```kotlin
// request a plugin
plugins {
    id("some.known.id") version "1.0"
}
```

```kotlin
// depend on a plugin
dependencies {
    implementation(plugin("some.known.id") version "1.0")
}
```

@JLLeitschuh yes, the Groovy DSL should get this added too.

**Comment by mkobit** [comment]
This should be possible now I think, but was going to try it either today or tomorrow. The `buildSrc` use case is the one I'm hoping to use right now.

**Comment by JLLeitschuh** [comment]
I was actually hoping to use exactly this in my `buildSrc` project later this month.

**Comment by mkobit** [comment]
I've been using `fun gradlePlugin(id: String, version: String): String = "$id:$id.gradle.plugin:$version"` which seems to be working for those that publish the marker artifact.

**Comment by gildor** [comment]
Why this task postponed to 2.0? It's a pretty useful feature and maybe simple extension function as in @mkobit comment would be helpful.

**Comment by LouisCAD** [comment]
How can I use @mkobit suggestion for the `maven-publish` gradle plugin in my `buildSrc`? I don't know the version and I'm unsure about the id.

**Comment by gildor** [comment]
`maven-publish` is core Gradle plugin and doesn't require any dependencies you already have it in your buildSrc classpath
